edgelet/edgelet-core/src/module.rs

```diff
@@ -323,16 +325,16 @@ impl SystemInfo {
     }
 }
 
-impl Default for SystemInfo {
-    fn default() -> Self {
-        let kernel = nix::sys::utsname::uname();
+impl SystemInfo {
+    pub fn new() -> nix::Result<Self> {
+        let kernel = nix::sys::utsname::uname()?;
         let dmi = DmiInfo::default();
         let os = OsInfo::default();
 
         let res = Self {
-            kernel: kernel.sysname().to_owned(),
-            kernel_release: kernel.release().to_owned(),
-            kernel_version: kernel.version().to_owned(),
+            kernel: kernel.sysname().to_str().unwrap_or("UNKNOWN").to_owned(),
+            kernel_release: kernel.release().to_str().map(ToOwned::to_owned),
+            kernel_version: kernel.version().to_str().map(ToOwned::to_owned),
```

Comment:
Why is the first one `"UNKNOWN"` on failure but the rest are skipped on failure? As in, why not pick one or the other?

Reply:
`kernel` maps to `osType`, which is a required field in the management API: e.g. https://github.com/Azure/iotedge/blob/f6af21abac0d5bcf30c1e2f917e03f18d061c0d0/edgelet/api/managementVersion_2021_12_07.yaml#L675. Had it not been a required field, I would have made `kernel` match the other fields.

Reply:
I added a comment regarding the special handling of `kernel` in d33a376c61bca7f28a7038ca22371fb142062f7d. However, I can still change the other fields to unwrap to `"UNKNOWN"` if that seems more sensible.

Reply:
Ah, that's okay then.